internal/usecase/crypto.go

```diff
@@ -156,6 +188,108 @@ func (uc *CryptoUsecase) GetDecryptionKey(ctx context.Context, identity string)
 	}, nil
 }
 
+func (uc *CryptoUsecase) GetDataForEncryption(ctx context.Context, address string, identityPrefixStringified string) (*GetDataForEncryptionResponse, *httpError.Http) {
+	if !ethCommon.IsHexAddress(address) {
+		log.Warn().Str("address", address).Msg("invalid address")
+		err := httpError.NewHttpError(
+			"invalid address",
+			"",
+			http.StatusBadRequest,
+		)
+		return nil, &err
+	}
+	var identityPrefix shcrypto.Block
+
+	if len(identityPrefixStringified) > 0 {
+		identityPrefixBytes, err := hex.DecodeString(strings.TrimPrefix(identityPrefixStringified, "0x"))
+		if err != nil {
+			log.Err(err).Msg("err encountered while decoding identity prefix")
+			err := httpError.NewHttpError(
+				"error encountered while decoding identity prefix",
+				"",
+				http.StatusBadRequest,
+			)
+			return nil, &err
+		}
+		if len(identityPrefixBytes) != 32 {
```

Comment:
we need to make check on the bytes and in order to get the bytes we need to decode it. We cant make a check on the strings since it wont give the correct length